akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala

```diff
@@ -141,6 +141,15 @@ abstract class ActorSystem[-T] extends ActorRef[T] with Extensions with ClassicA
    */
   def deadLetters[U]: ActorRef[U]
 
+  // this needs to be a lazy val because some impl of ActorSystem
+  // (eg: ActorSystemStub) does not have a provider
+  private lazy val cachedIgnoreRef: ActorRef[Nothing] = ActorRefAdapter(provider.ignoreRef)
```

Comment:
Builds were surprisingly failing all over the place because of ActorSystemStub. It's unfortunate that we need to make it lazy because of test concerns.

Reply:
And it would be unfortunate if ignoreRef can't be used with ActorSystemStub (synchronous behavior testing).
I think it would be better to override the `def ignoreRef` implementation in `ActorSystemStub` and return a DebugRef there, in the same way as `deadLetters`.

Reply:
regarding the lazy, this can be moved to `ActorSystemAdapter` and leave `def ignoreRef` unimplemented (abstract) in ActorSystem